Docs: Reporting restructure (#98913)
grafana/grafana: docs/sources/dashboards/create-reports/_index.md

  temp-data-lifetime:
    - pattern: /docs/grafana/
      destination: /docs/grafana/<GRAFANA_VERSION>/setup-grafana/configure-grafana/#temp_data_lifetime
    - pattern: /docs/grafana-cloud/
      destination: /docs/grafana/<GRAFANA_VERSION>/setup-grafana/configure-grafana/#temp_data_lifetime
  rendering-configuration:
    - pattern: /docs/grafana/
      destination: /docs/grafana/<GRAFANA_VERSION>/dashboards/create-reports/report-settings/#rendering-configuration
    - pattern: /docs/grafana-cloud/
      destination: /docs/grafana/<GRAFANA_VERSION>/dashboards/create-reports/report-settings/#rendering-configuration
  
---

# Create and manage reports

**Reporting** allows you to automatically generate PDFs from any of your dashboards and have Grafana email them to interested parties on a schedule. This is available in Grafana Cloud and in Grafana Enterprise.

{{< admonition type="note" >}}
If you have [Role-based access control](ref:role-based-access-control) enabled, for some actions you would need to have relevant permissions.
Refer to specific guides to understand what permissions are required.
{{< /admonition >}}

> Any changes you make to a dashboard used in a report are reflected the next time the report is sent. For example, if you change the time range in the dashboard, then the time range in the report also changes, unless you've configured a custom time range.

Reviewer: What you describe is the behavior when you create a report from a dashboard (and the URL got rewritten to include the timerange) but if you create a new report from the "Reports" page, time range is empty and the behavior in the doc is true.

## Requirements

- SMTP must be configured for reports to be sent. Refer to [SMTP](ref:smtp) in [Configuration](ref:configuration) for more information.
- The Image Renderer plugin must be installed or the remote rendering service must be set up. Refer to [Image rendering](ref:image-rendering) for more information.

## Access control

When [RBAC](ref:rbac) is enabled, you need to have the relevant [Permissions](ref:permission) to create and manage reports.

## Create a report

Only organization administrators can create reports by default. You can customize who can create reports with [Role-based access control](ref:role-based-access-control).

> The report creation process is multi-step, but you don't need to complete these steps in order and you can skip steps by clicking a step name at the top of the page:

Reviewer 2: The detail about steps is repeated below. Is it needed in both places?

   - [Select dashboard](#1-select-dashboard)
   - [Format report](#2-format-report)
   - [Schedule](#3-schedule)
   - [Share](#4-share)
   - [Confirm](#5-confirm)
1. Click one of the following buttons in the top-right corner of the screen:
   - **Send** or **Schedule send** - The report is sent according the schedule you've set.
   - **Save as draft** - You can save a draft at any point during the report creation or update process, even if it's missing required fields. The report won't be sent according to its schedule while it's a draft.
   - **Discard** - Delete the report draft. This action can't be reversed.

### 1. Select dashboard

At this step, select the dashboard or dashboards on which the report is based, as well as the variables and time range for those dashboards. The options are:

<!-- prettier-ignore-start -->

| Option | Description |
| ------ | ----------- |
| Source dashboard | Select the dashboard from which you want to generate the report. |
| [Template variables](#template-variables) | Select the variable **Datasource** and **Instance** for the selected dashboard. This option is only displayed if the dashboard has variables. |

> | [Time range (optional)](#time-range) | By default, reports use the saved time range of the dashboard. Optionally, you can change the time range of the report. |

Reviewer 2: I would mention here that in order to always use the dashboard time range, it needs to be blank by clicking the X at the end of the input since this is hard to find.

Reviewer: See my comment above, this would be prefilled only if you create a report from a dashboard. Maybe we should change the `By default` to say `If empty` so it's always true.

> | Add another dashboard | Add more dashboards to the report. Additional dashboards will be rendered as new pages in the same PDF file, or additional images if you chose to embed images in your report email. You can't add the same dashboard to a report multiple times. |

Reviewer 2: > You can’t add the same dashboard to a report multiple times

This is not true, a Report can contain the same dashboard more than once. One possible use case could be with different time ranges for example.

You can configure report-specific template variables for the dashboard on the report page. The variables that you select will override the variables from the dashboard, and they are used when rendering a PDF file of the report. For detailed information about using template variables, refer to the [Templates and variables](ref:templates-and-variables) section.

The query variables saved with a report might become out of date if the results of that query change. For example, if your template variable queries for a list of hostnames and a new hostname is added, then it will not be included in the report. If that occurs, the selected variables must be manually updated in the report. If you select the **All** value for the template variable or if you keep the dashboard's original variable selection, then the report stays up-to-date as new values are added.

##### Reports with panels or rows set to repeat by a variable

You can include dynamic dashboards with panels or rows, set to repeat by a variable, into reports. For detailed information about setting up repeating panels or rows in dashboards, refer to [Repeat panels or rows](ref:repeat-panels-or-rows).

> #### Time range

Reviewer 2: In this section, we should mention how to set a report always using the dashboard time range (the X I mentioned above).

| Simple | Portrait    | Generates an A4 page in portrait mode with three panels per page.                                         | {{< figure src="/static/img/docs/enterprise/reports_portrait_preview.png" max-width="500px" alt="Simple layout in portrait" >}}    |
| Simple | Landscape   | Generates an A4 page in landscape mode with a single panel per page.                                      | {{< figure src="/static/img/docs/enterprise/reports_landscape_preview.png" max-width="500px" alt="Simple layout in landscape" >}}                     |
| Grid   | Portrait    | Generates an A4 page in portrait mode with panels arranged in the same way as at the original dashboard.  | {{< figure src="/static/img/docs/enterprise/reports_grid_portrait_preview.png" max-width="500px" alt="Grid layout in portrait" >}} |
| Grid   | Landscape   | Generates an A4 page in landscape mode with panels arranged in the same way as in the original dashboard. | {{< figure src="/static/img/docs/enterprise/reports_grid_landscape_preview.png" max-width="500px" alt="Grid layout in landscape" >}}                  |

<!-- prettier-ignore-end -->

#### Embed a dashboard as an image in the email

> You can send a report email with an image of the dashboard embedded in the email instead of attached as a PDF. In this case, the email recipients can see the dashboard at a glance instead of having to open the PDF.

Reviewer: `instead of` suggest that it's one or the other while the user can select both PDF and image and will receive the embedded image with a PDF attachment.

#### Attach a CSV file of the table panel data

{{< admonition type="note" >}}
To use this feature in Grafana Enterprise, you must have [Grafana image renderer plugin](/grafana/plugins/grafana-image-renderer) v3.0.
{{< /admonition >}}

> You can attach a CSV file to the report email for each table panel on the selected dashboard, along with the PDF report. By default, CSVs larger than 10Mb are not sent which keeps email servers from rejecting the email. You can increase or decrease this limit in the [reporting configuration](ref:rendering-configuration).

Author: I didn't think that we wanted to avoid the plural of CSV so I opened a PR to add the pluralization as an option. I think this would be a good candidate for showing a suggestion for "CSV files" if someone writes "CSVs".

> You can attach a CSV file to the report email for each table panel on the selected dashboard, along with the PDF report. By default, CSVs larger than 10Mb are not sent which keeps email servers from rejecting the email. You can increase or decrease this limit in the [reporting configuration](ref:rendering-configuration).

Reviewer 3: Happy to add CSVs to the dictionary if we think it's cool

> You can attach a CSV file to the report email for each table panel on the selected dashboard, along with the PDF report. By default, CSVs larger than 10Mb are not sent which keeps email servers from rejecting the email. You can increase or decrease this limit in the [reporting configuration](ref:rendering-configuration).

Author: Already done! But when I tried rerunning the Vale job on this PR, it still failed. Not sure why?

> You can attach a CSV file to the report email for each table panel on the selected dashboard, along with the PDF report. By default, CSVs larger than 10Mb are not sent which keeps email servers from rejecting the email. You can increase or decrease this limit in the [reporting configuration](ref:rendering-configuration).

Author: Though, to be fair, CSV files is more accurate 🙃

> You can attach a CSV file to the report email for each table panel on the selected dashboard, along with the PDF report. By default, CSV files larger than 10Mb are not sent, which keeps email servers from rejecting the email. You can increase or decrease this limit in the [reporting configuration](ref:rendering-configuration).

Reviewer: > By default, CSV files larger than 10Mb are not sent

This is also true for images and PDFs so this could move into a section common to all formats, maybe in the "Rendering configuration" section in the settings page?

> You can attach a CSV file to the report email for each table panel on the selected dashboard, along with the PDF report. By default, CSV files larger than 10Mb are not sent, which keeps email servers from rejecting the email. You can increase or decrease this limit in the [reporting configuration](ref:rendering-configuration).

Reviewer: Should we mention that this will create one CSV file per table panel?

> This feature relies on the same plugin that supports the [image rendering](ref:image-rendering) features.

Reviewer: ```suggestion
```
This is true for every format and is already mentioned in the requirements section.

When the CSV file is generated, it is temporarily written to the `csv` folder in the Grafana `data` folder.

> A background job runs every 10 minutes and removes temporary CSV files. You can configure how long a CSV file should be stored before being removed by configuring the [temp-data-lifetime](ref:temp-data-lifetime) setting. This setting also affects how long a renderer PNG file should be stored.

Reviewer: This is also true for every format (images are stored in `png` and pdfs in `pdf`). This should be written somewhere else, maybe in the "Rendering configuration" section as well.

> Click **Download CSV** in the top-right corner of the screen to see the file.

Reviewer: ```suggestion
Click **Download CSV** in the top-right corner of the screen to download a zip file of the CSVs for your selected dashboard.
```

#### Table data in PDF

{{% admonition type="note" %}}
Available in public preview (`pdfTables` feature toggle) in [Grafana Enterprise](ref:grafana-enterprise) v10.3+ with the [Grafana image renderer plugin](/grafana/plugins/grafana-image-renderer) v3.0+, as well as [Grafana Cloud](/docs/grafana-cloud/).
{{% /admonition %}}

When there's more data in your table visualizations than can be shown in the dashboard PDF, you can select one of these two options to access all table visualization data as PDF in your reports:

> - **Include table data as PDF appendix** - Adds an appendix to the main dashboard PDF.

Reviewer: ```suggestion
- **Include table data as PDF appendix** - Adds an appendix to the dashboard PDF.
```

- **Attach a separate PDF of table data** - Generates a separate PDF file.

> This feature relies on the same plugin that supports the [image rendering](ref:image-rendering) features.

Reviewer: ```suggestion
```
This is true for every format.

| End date | If you leave the end date empty, the report is sent out indefinitely. |
| Send only from Monday to Friday | For reports that have an hourly or daily frequency, you can choose to send them only from Monday to Friday. |
| Send on the last day of the month | When you schedule a report with a monthly frequency, and set the start date between the 29th and the 31st of the month, the report is only sent during the months that have those dates. If you want the report to be sent every month, select this option instead. This way, the report is sent on the last day of every month regardless of how many days there are in any given month. |

<!-- prettier-ignore-end -->

### 4. Share

At this step, enter information related to sharing the report:

<!-- prettier-ignore-start -->

| Option | Description |
| ------ | ----------- |
| Report name | Name of the report as you want it to appear in the **Reports** list. The report name also populates the email subject line. |
| Recipients | Enter the email addresses of the people or teams that you want to receive the report, separated by commas or semicolons. |

> | Reply-to email address | The address that appears in the **Reply to** field of the email. |

Reviewer: ```suggestion
| Reply-to email address (optional) | The address that appears in the **Reply to** field of the email. |
```

| Message | Message body in the email with the report. |
| Include a dashboard link | Include a link to the dashboard from within the report email. |

<!-- prettier-ignore-end -->

> Click **Send test email** in the top-right corner of the screen to verify that the configuration works as expected and to verify that emails are working. You can choose to send this email to the recipients configured for the report, or to a different set of email addresses only used for testing. The last saved version of the report will be sent to selected emails.

Reviewer: ```suggestion
Click **Send test email** in the top-right corner of the screen to verify that the configuration works as expected and to verify that emails are working. You can choose to send this email to the recipients configured for the report, or to a different set of email addresses only used for testing.
```
This is not true anymore. If you make changes to the report and send a test email, it will include those even if you didn't save the report.

### 5. Confirm

At this step, the confirmation page displays all your the settings. Review them and confirm that they're correct or click the provided **Edit** links for each section to make updates.

Then, click **Send now** or **Schedule send**.

You can also save the report as a draft or discard it. Discarding the report is irreversible.

## Send a report using the API

You can send reports programmatically with the [send report](ref:send-report) endpoint in the [HTTP APIs](ref:http-apis).

> ## Manage reports

Reviewer: Should we add a `### Delete a report` section?

On the **Reports** page, you can view and manage your existing reports or create new ones.

![Reports page](/media/docs/grafana/dashboards/screenshot-reports-page-v11.5.png)

### Edit

To edit a report, follow these steps:

> 1. In the main menu, click **Dashboards > Reports**.

Reviewer: ```suggestion
1. In the main menu, click **Dashboards > Reporting**.
```

1. Click the row of the report you want to update.
1. Click the **Edit report** button in the top-right hand corner or click the **Edit** link for a specific section to go to that one directly.
1. When you've finished making changes, click **Confirm** at the top of the screen to go to the last step.
1. Click **Update report**.

### Pause or resume a report

You can pause and resume sending reports from the report list view. To do this, follow these steps:

> 1. In the main menu, click **Dashboards > Reports**.

Reviewer: ```suggestion
1. In the main menu, click **Dashboards > Reporting**.
```
